packages/flutter_tools/lib/src/hot.dart

```diff
@@ -238,8 +238,8 @@ class HotRunner extends ResidentRunner {
       if (connectionInfoCompleter != null) {
         connectionInfoCompleter.complete(
           new DebugConnectionInfo(
-            port: _observatoryPort,
-            wsUri: 'ws://localhost:$_observatoryPort/ws',
+            httpUri: _observatoryUri,
+            wsUri: vmService.wsAddress,
```

Comment:
I could do that, but then I'm making assumptions about the internals of the `VMService` class and breaking encapsulation. If you think it worth the effort, I can rework `DebugConnectionInfo` and `VMService` so that `VMService` returns a `DebugConnectionInfo` or something like that so as not to break encapsulation, but that's more involved than I want to tackle in this PR.